rocAL/rocAL/source/parameters/parameter_rocal_crop.cpp

```diff
@@ -27,56 +27,53 @@ THE SOFTWARE.
-            (crop_w > in_width[img_idx]) ? (cropw_arr_val[img_idx] = in_width[img_idx]) : (cropw_arr_val[img_idx] = crop_w);
-            (crop_h > in_height[img_idx]) ? (croph_arr_val[img_idx] = in_height[img_idx]) : (croph_arr_val[img_idx] = crop_h);
-            (x1 >= in_width[img_idx]) ? (x1_arr_val[img_idx] = 0) : (x1_arr_val[img_idx] = x1);
-            (y1 >= in_height[img_idx]) ? (y1_arr_val[img_idx] = 0) : (y1_arr_val[img_idx] = y1);
-        }
-        else
-        {
-            float crop_h_factor_, crop_w_factor_, x_drift, y_drift;
-            crop_height_factor->renew();
-            crop_h_factor_ = crop_height_factor->get();
-            crop_width_factor->renew();
-            crop_w_factor_ = crop_width_factor->get();
-            cropw_arr_val[img_idx] = static_cast<size_t> (crop_w_factor_ * in_width[img_idx]);
-            croph_arr_val[img_idx] = static_cast<size_t> (crop_h_factor_ * in_height[img_idx]);
-            x_drift_factor->renew();
-            y_drift_factor->renew();
-            y_drift_factor->renew();
-            x_drift = x_drift_factor->get();
-            y_drift = y_drift_factor->get();
-            x1_arr_val[img_idx] = static_cast<size_t>(x_drift * (in_width[img_idx]  - cropw_arr_val[img_idx]));
-            y1_arr_val[img_idx] = static_cast<size_t>(y_drift * (in_height[img_idx] - croph_arr_val[img_idx]));
+            cropw_arr_val[img_idx] = (crop_w <= in_width[img_idx] && crop_w > 0) ? crop_w : in_width[img_idx];
+            croph_arr_val[img_idx] = (crop_h <= in_height[img_idx] && crop_h > 0) ? crop_h : in_height[img_idx];
+            if (_is_center_crop) {
+                x1_arr_val[img_idx] = static_cast<size_t>(x_drift_factor->get() * (in_width[img_idx] - cropw_arr_val[img_idx]));
+                y1_arr_val[img_idx] = static_cast<size_t>(y_drift_factor->get() *(in_height[img_idx] - croph_arr_val[img_idx]));
+            } else {
+              x1_arr_val[img_idx] = (x1 >= in_width[img_idx]) ? 0 : x1;
+              y1_arr_val[img_idx] = (y1 >= in_height[img_idx]) ? 0 : y1;
+            }
+        } else {
+          float crop_h_factor_, crop_w_factor_, x_drift, y_drift;
+          crop_height_factor->renew();
+          crop_h_factor_ = crop_height_factor->get();
+          crop_width_factor->renew();
+          crop_w_factor_ = crop_width_factor->get();
+          cropw_arr_val[img_idx] = static_cast<size_t> (crop_w_factor_ * in_width[img_idx]);
+          croph_arr_val[img_idx] = static_cast<size_t> (crop_h_factor_ * in_height[img_idx]);
+          x_drift_factor->renew();
+          y_drift_factor->renew();
+          y_drift_factor->renew();
+          x_drift = x_drift_factor->get();
+          y_drift = y_drift_factor->get();
+          x1_arr_val[img_idx] = static_cast<size_t>(x_drift * (in_width[img_idx]  - cropw_arr_val[img_idx]));
+          y1_arr_val[img_idx] = static_cast<size_t>(y_drift * (in_height[img_idx] - croph_arr_val[img_idx]));
```

Comment:
We need to check this algorithm once. 
Wherever possible change conditional operator and if condition to std::min or max. 
Are we checking if x+crop_width <= im_width?